frameworklauncher/src/main/java/com/microsoft/frameworklauncher/service/StatusManager.java

```diff
@@ -287,6 +288,56 @@ private void upgradeFramework(FrameworkRequest frameworkRequest) throws Exceptio
     addFramework(frameworkRequest);
   }
 
+  private void updateFramework(FrameworkRequest frameworkRequest) throws Exception {
+    String frameworkName = frameworkRequest.getFrameworkName();
+    FrameworkStatus frameworkStatus = frameworkStatuses.get(frameworkName);
+    FrameworkState frameworkState = frameworkStatus.getFrameworkState();
+
+    if (FrameworkStateDefinition.FINAL_STATES.contains(frameworkState)) {
+      // Ignore to updateFramework since the Framework of current Version is already in FINAL_STATES,
+      // such as the Framework completed by itself or completed by previous ExecutionType.STOP.
+      // And Framework Status in FINAL_STATES should be immutable.
+      return;
+    }
+
+    updateExecutionType(frameworkRequest);
+  }
+
+  private void updateExecutionType(FrameworkRequest frameworkRequest) throws Exception {
+    String frameworkName = frameworkRequest.getFrameworkName();
+    ExecutionType executionType = frameworkRequest.getFrameworkDescriptor().getExecutionType();
+
+    if (executionType == ExecutionType.STOP) {
```

Comment:
>STOP [](start = 39, length = 4)

STOP is the only executionType handled in this function. do you plan to extend the executionType later on?

Reply:
maybe extend it.

---
In reply to: [171839789](https://github.com/Microsoft/pai/pull/346#discussion_r171839789) [](ancestors = 171839789)